### PR TITLE
Moodle 4.1 fixes

### LINKED ISSUE #50: Test support for 4.1 -> Some test failures found
I have tested installation and automated tests on Moodle 4.1. I found the following minor issues:

- The unit test classes need renaming, as PHPUnit has deprecated class name that do not match the file name
- Unrelated to 4.1 but still worth fixing, a test fails if the development environment has a proxy configured.

### PR DESCRIPTION
**Pull request by marxjohnson**
This resolves #50 and should confirm the plugin works with Moodle 4.1, although there may be issues found when testing a plugin that depends on this (e.g. objectfs).

### CODE REVIEW

**Comment by marxjohnson** [comment]
I have also tested up and updated tool_objectfs on 4.1 using this branch, and found no further issues.

File: version.php [hunk 1]
```diff
@@ -30,4 +30,4 @@
 $plugin->requires  = 2013111811;
 $plugin->component = 'local_aws';
 $plugin->maturity  = MATURITY_STABLE;
-$plugin->supported = [26, 311]; // A range of branch numbers of supported moodle versions.
+$plugin->supported = [26, 401]; // A range of branch numbers of supported moodle versions.
```
Comment (brendanheywood) [inline version.php:33]: @marxjohnson this is good to go, can you please bump the version and then merge away

**Comment by marxjohnson** [comment]
Thanks @brendanheywood, I've bumped the version but I don't have write access to this repo so please complete the merge for me when you can.

**Comment by brendanheywood** [comment]
@marxjohnson your access should now be sorted

**Comment by cameron1729** [comment]
b250fd8f introduced these test failures - yet it got the green tick: https://github.com/catalyst/moodle-local_aws/actions/runs/3870967083

Could we update CI as part of this PR?

**Comment by marxjohnson** [comment]
@cameron1729 Ok I have updated the unit tests and made sure CI tests are running up to Moodle 4.1. I'll go ahead with the merge now.

**Pull request merged**